contracts/SnappBase.sol

```diff
@@ -80,22 +91,30 @@ contract SnappBase is Ownable {
             "Unsuccessful transfer"
         );
 
-        uint depositSlot = depositSlot();
-        if (depositHashes[depositSlot].shaHash == bytes32(0)) {
-            slotIndex = 0;
+        // uint depositIndex = deposits.length - 1;
+        DepositState memory currDepositState = deposits[depositIndex];
+        if (currDepositState.size == MAX_DEPOSIT_BATCH || block.number > currDepositState.creationBlock + 20) {
+            currDepositState = DepositState({
+                size: 0,
+                shaHash: bytes32(0),
+                creationBlock: block.number,
+                appliedAccountStateIndex: 0   // Default 0 implies not applied.
+            });
+            depositIndex++;
+            deposits[depositIndex] = currDepositState;
         }
+
+        // Update Deposit Hash based on request
         uint16 accountId = publicKeyToAccountMap[msg.sender];
         bytes32 nextDepositHash = sha256(
-            abi.encodePacked(depositHashes[depositSlot].shaHash, accountId, tokenIndex, amount)
+            abi.encodePacked(currDepositState.shaHash, accountId, tokenIndex, amount)
         );
-        depositHashes[depositSlot] = DepositState({shaHash: nextDepositHash, applied: false});
+        currDepositState.shaHash = nextDepositHash;
+        currDepositState.size++;
 
-        emit Deposit(accountId, tokenIndex, amount, depositSlot, slotIndex);
-        slotIndex++;
-    }
+        deposits[depositIndex] = currDepositState;
```

Comment:
Is this actually necessary? I know it was for the array... but might not be for the mapping (since we were mutating what the mapping was already pointing at).